database/sql/src/main/scala/cromwell/database/slick/tables/GroupMetricsEntryComponent.scala

```diff
@@ -0,0 +1,34 @@
+package cromwell.database.slick.tables
+
+import cromwell.database.sql.tables.GroupMetricsEntry
+import slick.lifted.ProvenShape
+
+import java.sql.Timestamp
+
+trait GroupMetricsEntryComponent {
+
+  this: DriverComponent =>
+
+  import driver.api._
+
+  class GroupMetricsEntries(tag: Tag) extends Table[GroupMetricsEntry](tag, "GROUP_METRICS_ENTRY") {
+
+    def groupMetricsEntryId = column[Long]("GROUP_METRICS_ENTRY_ID", O.PrimaryKey, O.AutoInc)
+
+    def groupId = column[String]("GROUP_ID", O.Length(255))
+
+    def quotaExhaustionDetected = column[Timestamp]("QUOTA_EXHAUSTION_DETECTED")
+
+    override def * : ProvenShape[GroupMetricsEntry] = (groupId,
+                                                       quotaExhaustionDetected,
+                                                       groupMetricsEntryId.?
+    ) <> ((GroupMetricsEntry.apply _).tupled, GroupMetricsEntry.unapply)
```

Comment:
Question for reviewers - For default projection method `*`, I added `.?` at the end of  `groupMetricsEntryId` since the `...Id` columns in other EntryComponents also do that. As a result the `groupMetricsEntryId` in `GroupMetricsEntry.scala` is also marked as Optional[Long]. My understanding is that `?` lifts a column to optional but I am not sure why an id column which is the primary key should be optional?

Reply:
Discussed in meeting, and added a comment explaining the reason in https://github.com/broadinstitute/cromwell/pull/7487/commits/a9989766d77e0d6f05dc193552addad1a83dc0a7.